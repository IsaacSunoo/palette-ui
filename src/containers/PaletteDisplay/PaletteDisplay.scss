.PaletteDisplay {
  display: flex;
  height: 84%;

  @media screen and (max-width: $break-large) {
    flex-direction: column;
  }

}
