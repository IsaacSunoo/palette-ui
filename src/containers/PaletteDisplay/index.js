import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// import { Loading } from '../../components/Loading';
import Color from '../Color'

export class PaletteDisplay extends Component {

  render(){
    const { currentPalette } = this.props;

    return (
      <div className="PaletteDisplay">
        {
          currentPalette.length &&
          currentPalette.map((color, index) => {
            return <Color key={color.color} {...color} id={index}/>
          })
        }
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({
  currentPalette: state.currentPalette
})

export const mapDispatchToProps = (dispatch) => ({

})

PaletteDisplay.propTypes = {
  currentPalette: PropTypes.array,
}

export default connect(mapStateToProps, mapDispatchToProps)(PaletteDisplay);
