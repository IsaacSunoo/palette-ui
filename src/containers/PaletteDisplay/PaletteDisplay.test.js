import React from 'react';
import { shallow } from 'enzyme';
import { PaletteDisplay, mapStateToProps, mapDispatchToProps } from './PaletteDisplay';

describe('PaletteDisplay', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PaletteDisplay />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
