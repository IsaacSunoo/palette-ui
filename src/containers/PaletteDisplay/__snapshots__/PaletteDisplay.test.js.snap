// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteDisplay PaletteDisplay Component should match the snapshot 1`] = `
<div
  className="PaletteDisplay"
>
  <ConnectFunction
    color="#fffff"
    id={0}
    key="#fffff"
    locked={false}
  />
  <ConnectFunction
    color="#00000"
    id={1}
    key="#00000"
    locked={false}
  />
  <ConnectFunction
    color="#231f20"
    id={2}
    key="#231f20"
    locked={false}
  />
  <ConnectFunction
    color="#e6e6e6"
    id={3}
    key="#e6e6e6"
    locked={false}
  />
  <ConnectFunction
    color="#cccccc"
    id={4}
    key="#cccccc"
    locked={false}
  />
</div>
`;
